### ISSUE TITLE
Equals extensions

### ISSUE THREAD

**Issue by Renesanse** [leisim/dartx#52]
Please add equals extensions based on ListEquality and so on...

**Comment by leisim** [comment]
It is not possible to create an extension named `equals` because every object already has a method with that name. There already exists the `.contentEquals()` extension for iterables (and lists).

**Comment by passsy** [comment]
I like the idea but I'm unsure if we should use [`ListEquality`](https://api.flutter.dev/flutter/package-collection_collection/ListEquality-class.html) or [`DeepCollectionEquality`](https://api.flutter.dev/flutter/package-collection_collection/DeepCollectionEquality/DeepCollectionEquality.html). Please describe your use case and we can see which API fits best.

Alternatively you could use collections which support equality by default: [`BuildList`](https://pub.dev/documentation/built_collection/latest/built_collection/BuiltList-class.html) or [`KtList`](https://pub.dev/documentation/kt_dart/latest/kt_collection/KtList-class.html)

**Comment by Renesanse** [comment]
You probably mean == operator, I didn't see a equals in Iterable, List, etc.  ```.contentEquals()``` is ok.

**Comment by leisim** [comment]
> You probably mean == operator

Yes, my bad. Is the `contentEquals()` method not suggested for auto-complete if you type "equals"?

**Comment by Renesanse** [comment]
To be honest I am against about to use different collections. I prefer use core lib with extensions =)

**Comment by leisim** [comment]
Does `contentEquals()` solve your problem or do you need more options / deep collection equality?

**Comment by Renesanse** [comment]
Well, contentEquals solved my problem, but I think will be nice to have deep collection equality as well.
